addons/web_editor/static/src/js/wysiwyg/widgets/upload_progress_toast.js

```diff
@@ -14,62 +12,34 @@ export const UploadProgressToast = Widget.extend({
         this._super(...arguments);
-        this.resId = resId;
-        this.resModel = resModel;
-        this.isImage = isImage;
-
-        // Upload the smallest file first to block the user the least possible.
-        this.files = [...files].sort((a, b) => a.size - b.size);
-        this.files.forEach((file, index) => {
-            file.id = index;
-            if (!file.size) {
-                file.displaySize = null;
-            } else if (file.size < 1024) {
-                file.displaySize = file.size.toFixed(2) + " bytes";
-            } else if (file.size < 1048576) {
-                file.displaySize = (file.size / 1024).toFixed(2) + " KB";
-            } else {
-                file.displaySize = (file.size / 1048576).toFixed(2) + " MB";
-            }
-        });
+        this.files = files;
     },
     /**
      * @override
      */
     start() {
-        this.hasError = false;
-        const uploadMutex = new concurrency.Mutex();
-        this.files.forEach((file, index) => {
-            // Upload one file at a time: no need to parallel as upload is
-            // limited by bandwidth.
-            uploadMutex.exec(async () => {
-                const dataURL = await getDataURLFromFile(file);
-                const attachment = await this._rpcShowProgress({
-                    route: '/web_editor/attachment/add_data',
-                    params: {
-                        name: file.name,
-                        data: dataURL.split(',')[1],
-                        res_id: this.resId,
-                        res_model: this.resModel,
-                        is_image: this.isImage,
-                        width: 0,
-                        quality: 0,
-                    }
-                }, this.$el.find(`.js_progressbar_${index}`));
-                if (!attachment.error) {
-                    this.trigger_up('file_complete', attachment);
-                }
-            });
-        });
+        this.$progress = $('<div/>');
+        _.each(this.files, (file, index) => {
+            let fileSize = file.size;
+            if (!fileSize) {
+                fileSize = null;
+            } else if (fileSize < 1024) {
+                fileSize = fileSize.toFixed(2) + " bytes";
+            } else if (fileSize < 1048576) {
+                fileSize = (fileSize / 1024).toFixed(2) + " KB";
+            } else {
+                fileSize = (fileSize / 1048576).toFixed(2) + " MB";
+            }
 
-        uploadMutex.getUnlockedDef().then(() => {
-            this.trigger_up('upload_complete');
-            this.destroyOnClose = true;
-            this.close(2000);
+            this.$progress.append(qweb.render('wysiwyg.widgets.upload.progressbar', {
+                fileId: index,
+                fileName: file.name,
+                fileSize: fileSize,
+            }));
         });
-        this.uploadMutex = uploadMutex;
+        this.$el.find('.o_notification_content').html(this.$progress);
```

Comment:
`.html(` makes the PR red (security check). In this case, it's not really perfect to call it anyway since you already have a jQuery element: `.empty().append(...)` (if empty actually necessary?)

Reply:
@rdeodoo I did not see https://github.com/odoo/odoo/pull/76360#issuecomment-919319703 before :point_up: Still: only use `.html(` (or `.innerHTML =`) when you are actually adding a string containing HTML (here you have nodes / jquery elements).

Reply:
> only use .html( (or .innerHTML =) when you are actually adding a string containing HTML (here you have nodes / jquery elements).

Didn't know that, but indeed, the jQuery doc agrees. (even if I guess they accept jQuery element and not only string)